pkg/fanal/analyzer/language/nodejs/yarn/yarn.go

```diff
@@ -211,11 +218,16 @@ func (a yarnAnalyzer) walkDependencies(pkgs []types.Package, pkgIDs map[string]t
 			constraint = m[4]
 		}
 
-		// npm has own comparer to compare versions
-		if match, err := a.comparer.MatchVersion(pkg.Version, constraint); err != nil {
-			return nil, xerrors.Errorf("unable to match version for %s", pkg.Name)
-		} else if !match {
-			continue
+		// Try to find an exact match to the pattern.
+		// In some cases, patterns from yarn.lock and package.json may not match (e.g., yarn v2 uses the allowed version for ID).
```

Comment:
>(e.g., yarn v2 uses the allowed version for ID).

Could you share the actual example? I didn't know there are cases where they don't match.

Reply:
I saw this when I updated tests:
1.  [package.json with "^3.0.1"](https://github.com/aquasecurity/trivy/blob/805592d7ec3d4bc55982ab51e8533d5a254e8853/pkg/fanal/analyzer/language/nodejs/yarn/testdata/yarn-licenses/package.json#L7) + [yarn.lock with "3.0.1"](https://github.com/aquasecurity/trivy/blob/805592d7ec3d4bc55982ab51e8533d5a254e8853/pkg/fanal/analyzer/language/nodejs/yarn/testdata/yarn-licenses/yarn.lock#L22)

Reply:
It looks really weird to me. They should match. But anyway, it's good to fall back into version range matching.